src/components/PostsList.tsx

```diff
@@ -1,86 +1,72 @@
 import React from 'react';
+import { Post } from '../types/Post';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  userPosts: Post[];
+  handleShowComment: (post: Post) => void;
+  showSideBar: number | null;
+  setShowSideBar: (id: number | null) => void;
+  setShowCommentForm: (item: boolean) => void;
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = ({
+  userPosts,
+  handleShowComment,
+  showSideBar,
+  setShowSideBar,
+  setShowCommentForm,
+}) => {
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
 
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <th> </th>
+          </tr>
+        </thead>
 
-          <td data-cy="PostTitle">
-            fugit voluptas sed molestias voluptatem provident
-          </td>
+        <tbody>
+          {userPosts.map((post: Post) => (
```

Comment:
It's not recommended to generate keys on render, because it can lead to unexpected behavior. In this case, the key is generated from the post id, which is a good practice since it is unique for each post. But, as a general rule, avoid generating keys on render.